### PR TITLE
Refresh Visual Studio Edge debugging docs

### LINKED ISSUE #1350: What VS project template is the one used here?
As far as I can tell, there is no project template that magically gives you the project used to illustrate this article.  

The one shown in the article is described as including pretty much every possible variation.  Meanwhile, the actual templates provided (VS 2019 Enterprise 16.10.3) are specific to a particular target i.e. React, Angular, MVC, Web API.

So this ends up being another 'all you have to do' that turns into a major time suck.  It's really annoying trying to get my example to match your example.  Maybe provide your Project for download because I see nothing that doesn't involve an hour or two of messing around. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4538b351-b450-e104-3f5c-76bd9c5233e9
* Version Independent ID: 700de4b5-e02b-8914-c0a6-0ab604025e1f
* Content: [Visual Studio - Microsoft Edge Development](https://docs.microsoft.com/en-us/microsoft-edge/visual-studio/)
* Content Source: [microsoft-edge/visual-studio/index.md](https://github.com/MicrosoftDocs/edge-developer/blob/master/microsoft-edge/visual-studio/index.md)
* Product: **microsoft-edge**
* GitHub Login: @MSEdgeTeam
* Microsoft Alias: **msedgedevrel**

### PR DESCRIPTION
**Pull request by captainbrosset**
The project template to be used in VS 2019 changed so the documentation wasn't correct anymore. The wording had to be changed and so did the screenshot.

The UI of the react.js webapp also changed, so the screenshots had to beupdated too.

I also changed some of the wording as per the MWSG.

Fixes #1350

### CODE REVIEW

File: microsoft-edge/visual-studio/index.md [hunk 1]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
 
-Provide a **Project name** for your new project and choose **Create**.  For the purposes of the example, choose **React.js** as the template, and choose **Create**.  The **React.js** template specifies how to integrate React.js with an ASP.NET Core app.  
+:::image type="complex" source="./media/create-new-project.png" alt-text="Create a new ASP.NET Core Web app with React.js" lightbox="./media/create-new-project.png":::
+   Create a new ASP.NET Core Web app with React.js  
+:::image-end:::  
 
 ### Launch Microsoft Edge from Visual Studio  
 
-After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, choose the dropdown next to the green **Play** button and **IIS Express**.  
+After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:47]:
```suggestion
1.  After you create your project, open `ClientApp/src/components/Counter.js`.

1.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.  
```
For closely related (conceptually grouped) user-actions that feel like a single conceptual step, group into a single numbered list item.  Avoid ultra-granular, fragmented numbered steps.
blank line spreads numbered steps apart, for readability.  
For the image block below, prefix spaces to move it right so it starts in column 4.  I think most Markdown, including this build system, treats column 4 as meaningful/significant; have to check the Doc Contributor Guide later.

File: microsoft-edge/visual-studio/index.md [hunk 2]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
+:::image-end:::  
 
 ### Launch Microsoft Edge from Visual Studio  
 
-After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, choose the dropdown next to the green **Play** button and **IIS Express**.  
+After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.  
 
 :::image type="complex" source="./media/vs-dropdown.png" alt-text="The dropdown next to the green Play button and IIS Express" lightbox="./media/vs-dropdown.png":::
    The dropdown next to the green **Play** button and **IIS Express**  
 :::image-end:::
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:51]:
```suggestion
    :::image type="complex" source="./media/vs-dropdown.png" alt-text="The dropdown next to the green Play button and IIS Express" lightbox="./media/vs-dropdown.png":::
       The dropdown next to the green **Play** button and **IIS Express**  
    :::image-end:::  
```
Start in col 4(?) to indent image and maybe to make the numbered steps below auto-increment correctly.

File: microsoft-edge/visual-studio/index.md [hunk 3]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
+Now that Visual Studio is correctly configured, select the green **Play** button.  Visual Studio builds your app, starts the web server, launches Microsoft Edge, and navigates to `https://localhost:44362/` or whatever port is specified in `launchSettings.json`.  
 
 :::image type="complex" source="./media/edge-launch.png" alt-text="Microsoft Edge launches from Visual Studio" lightbox="./media/edge-launch.png":::
    Microsoft Edge launches from Visual Studio  
 :::image-end:::  
 
 ### Debug JavaScript running in Microsoft Edge  
 
-Switch back to Visual Studio.  In `Counter.js`, to set a breakpoint on Line 13, choose the gutter next to the line.  
+Switch back to Visual Studio.  In `Counter.js`, to set a breakpoint on Line 13, select the gutter next to the line.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:73]: Depending on how thorough of an update this PR aims for, test the steps, make sure it's still Line 13.  Testing takes a long time, but can yield corrections.  But consider/balance/weigh the opportunity cost of such in-depth testing of steps.

File: microsoft-edge/visual-studio/index.md [hunk 4]
```diff
@@ -115,13 +112,13 @@ From the **Attach to Process** dialog, set **Connection type** to **Chrome devto
    Configure the **Attach to Process** dialog in Visual Studio  
 :::image-end:::  
 
-Choose **Select...** > the checkbox next to **JavaScript (Microsoft Edge – Chromium)**.  To add tabs, navigate to new tabs, and close tabs and display the changes reflected in the **Attach to Process** dialog, choose the **Refresh** button.  Choose the tab you want to debug and choose **Attach**.  
+Select **Select** > the checkbox next to **JavaScript (Microsoft Edge – Chromium)**.  To add tabs, navigate to new tabs, and close tabs and display the changes reflected in the **Attach to Process** dialog, select the **Refresh** button.  Choose the tab you want to debug and select **Attach**.  
 
 The Visual Studio debugger is now attached to Microsoft Edge.  You may pause the running of JavaScript, set breakpoints, and review `console.log()` statements directly in the Debug Output window in Visual Studio.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:117]: Global: check 'can' and 'may'.  Change 'may' to 'can' in most places.  Per MWSG's [can, may](https://styleguides.azurewebsites.net/Styleguide/Read?id=2700&topicid=33430) article: "When ability is what you need to express, it's OK to use can to describe actions or tasks that the reader or program is able to do. Use might to express possibility. **Don't use may**, which might be interpreted as providing permission."

Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:117]:
```suggestion
The Visual Studio debugger is now attached to Microsoft Edge.  You can pause the running of JavaScript, set breakpoints, and review `console.log()` statements directly in the Debug Output window in Visual Studio.  
```
I'd keep this left-justified.  Optionally shift w/ leading spaces to start in column 4(?) if it looks more consistent.

File: microsoft-edge/visual-studio/index.md [hunk 5]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
 
-Choose **Script Debugging** > **Enabled**.  
+Select **Script Debugging** > **Enabled**.  
 
 :::image type="complex" source="./media/enable-script-debugging.png" alt-text="Turn on script debugging in Visual Studio" lightbox="./media/enable-script-debugging.png":::
    Turn on script debugging in Visual Studio  
 :::image-end:::  
 
-In the same dropdown, choose **Web Browser** > the preview channel of Microsoft Edge that you want Visual Studio to launch, such as Microsoft Edge Canary, Dev, or Beta.  If you are not already using one of the Microsoft Edge preview channels, navigate to [Download Microsoft Edge Insider Channels][MicrosoftedgeinsiderDownload] to download one.  
+In the same dropdown, select **Web Browser** > the preview channel of Microsoft Edge that you want Visual Studio to launch, such as Microsoft Edge Canary, Dev, or Beta.  If you are not already using one of the Microsoft Edge preview channels, navigate to [Download Microsoft Edge Insider Channels][MicrosoftedgeinsiderDownload] to download one.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:59]: Elsewhere, change:
you may step through the script line-by-line
to:
you can step through the script line-by-line

File: microsoft-edge/visual-studio/index.md [hunk 6]
```diff
@@ -95,18 +92,18 @@ The example was just a minor demonstration of the functionality available in Vis
...
-From Visual Studio, open the **Debug** menu and choose **Attach to Process** or select `Ctrl`+`Alt`+`P`.  
+From Visual Studio, select **Debug** > **Attach to Process** or `Ctrl`+`Alt`+`P`.  
 
-:::image type="complex" source="./media/attach-to-process.png" alt-text="Choose Attach to Process in Visual Studio" lightbox="./media/attach-to-process.png":::
-   Choose **Attach to Process** in Visual Studio  
+:::image type="complex" source="./media/attach-to-process.png" alt-text="Select Attach to Process in Visual Studio" lightbox="./media/attach-to-process.png":::
+   Select **Attach to Process** in Visual Studio  
 :::image-end:::  
 
 From the **Attach to Process** dialog, set **Connection type** to **Chrome devtools protocol websocket (no authentication)**.  In the **Connecting target** textbox, type `http://localhost:9222/` and select `Enter`.  Review the list of open tabs you have in Microsoft Edge listed out in the **Attach to Process** dialog.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:109]:
```suggestion
1.  In the **Attach to Process** dialog, set **Connection type** to **Chrome devtools protocol websocket (no authentication)**.  
1.  In the **Connecting target** textbox, type `http://localhost:9222/` and select `Enter`.  In the **Attach to Process** dialog, review the list of tabs that are open in Microsoft Edge.  
```
Break into 2-3 cohesive steps.  Reading this step out of context, it's unclear how many dialog boxes are involved.  Each dialog would be a separate step.

File: microsoft-edge/visual-studio/index.md [hunk 7]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
+Switch back to Visual Studio.  In `Counter.js`, to set a breakpoint on Line 13, select the gutter next to the line.  
 
-:::image type="complex" source="./media/set-breakpoint.png" alt-text="Choose the gutter next to Line 13 in Counter.js to set a breakpoint in Visual Studio" lightbox="./media/set-breakpoint.png":::
-   Choose the gutter next to Line 13 in `Counter.js` to set a breakpoint in Visual Studio  
+:::image type="complex" source="./media/set-breakpoint.png" alt-text="Select the gutter next to Line 13 in Counter.js to set a breakpoint in Visual Studio" lightbox="./media/set-breakpoint.png":::
+   Select the gutter next to Line 13 in `Counter.js` to set a breakpoint in Visual Studio  
 :::image-end:::  
 
-Now switch back to the instance of Microsoft Edge that Visual Studio launched.  Choose on **Counter** in the NavMenu on the left of the webpage.  Now choose **Increment**.  
+Now switch back to the instance of Microsoft Edge that Visual Studio launched.  Select **Counter** in the navigation menu on the left of the webpage.  Now select **Increment**.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:79]:
```suggestion
1.  Switch back to the instance of Microsoft Edge that Visual Studio launched.

1.  Select **Counter** in the navigation menu at the top of the webpage, and then select the **Increment** button.
```
Check the location of "Counter" UI, in context.    Looks like it's in the upper right in the screenshot.
Style guide says "don't state the UI control type, unless it helps clarity".  My retort is, it _always_ helps clarity, esp given the Accessibility rule, "don't assume the screenshots are visible to the audience".  I omit "the ... button" only in the easy case of the OK button.

File: microsoft-edge/visual-studio/index.md [hunk 8]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
 
-Provide a **Project name** for your new project and choose **Create**.  For the purposes of the example, choose **React.js** as the template, and choose **Create**.  The **React.js** template specifies how to integrate React.js with an ASP.NET Core app.  
+:::image type="complex" source="./media/create-new-project.png" alt-text="Create a new ASP.NET Core Web app with React.js" lightbox="./media/create-new-project.png":::
+   Create a new ASP.NET Core Web app with React.js  
+:::image-end:::  
 
 ### Launch Microsoft Edge from Visual Studio  
 
-After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, choose the dropdown next to the green **Play** button and **IIS Express**.  
+After you create your project, open `ClientApp/src/components/Counter.js`.  Now, to use Visual Studio to debug JavaScript, select the dropdown next to the green **Play** button and **IIS Express**.
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:48]: near top of article, change:
Visual Studio completes the following workflow using a single button.
to:
Visual Studio completes the entire following workflow when you select a single button.
(which button? does the surrounding text fill-in a mental picture adequately to follow the meaning here?  i'd try to slip-in a hint of what we have in mind)

**Review by mikehoffms — CHANGES_REQUESTED** [review]

File: microsoft-edge/visual-studio/index.md [hunk 9]
```diff
@@ -34,52 +34,49 @@ The simplified workflow allows you to debug JavaScript that run in Microsoft Edg
...
-Open Visual Studio 2019 and choose **Create a new project**.  On the next screen, choose **ASP.NET Core Web app** > **Next**.  
+Open Visual Studio 2019 and select **Create a new project**.  In the search box on the next screen, enter **react**.  Select **ASP.NET Core with React.js** from the list of templates and then **Next**.  
 
-:::image type="complex" source="./media/create-new-project.png" alt-text="Create a new ASP.NET Core Web app" lightbox="./media/create-new-project.png":::
-   Create a new ASP.NET Core Web app  
-:::image-end:::  
+This **React.js** template specifies how to integrate React.js with an ASP.NET Core app.  
 
-Provide a **Project name** for your new project and choose **Create**.  For the purposes of the example, choose **React.js** as the template, and choose **Create**.  The **React.js** template specifies how to integrate React.js with an ASP.NET Core app.  
+:::image type="complex" source="./media/create-new-project.png" alt-text="Create a new ASP.NET Core Web app with React.js" lightbox="./media/create-new-project.png":::
```
Comment (mikehoffms) [inline microsoft-edge/visual-studio/index.md:41]: If not too much trouble/time, in screenshots, change yellow box to red.  Usually 3px not 5px, but that width could be hard to change.  
To change yellow to red in Paint:
1. Shift+click an image link in the .md file in VS Code.
2. Use a menu item or Sh+Alt+R to open Explorer at the dir.
3. Right-click an image file, Edit, to open in Paint.
4. Select or enter a red color.  Or open an image with red box and use Dropper tool to sample it.
4. Select Paint Bucket tool, click a yellow rectangle.  Save.

File: microsoft-edge/visual-studio/index.md [hunk 10]
```diff
@@ -115,13 +112,13 @@ From the **Attach to Process** dialog, set **Connection type** to **Chrome devto
    Configure the **Attach to Process** dialog in Visual Studio  
 :::image-end:::  
 
-Choose **Select...** > the checkbox next to **JavaScript (Microsoft Edge – Chromium)**.  To add tabs, navigate to new tabs, and close tabs and display the changes reflected in the **Attach to Process** dialog, choose the **Refresh** button.  Choose the tab you want to debug and choose **Attach**.  
+Select **Select** > the checkbox next to **JavaScript (Microsoft Edge – Chromium)**.  To add tabs, navigate to new tabs, and close tabs and display the changes reflected in the **Attach to Process** dialog, select the **Refresh** button.  Choose the tab you want to debug and select **Attach**.
```
[the remark this answers is not in the document]
Comment (captainbrosset) [reply microsoft-edge/visual-studio/index.md:115]: Thanks for catching this. I was mostly focusing on the first part of the page (`Debug JavaScript running in Microsoft Edge`) and didn't realize that (`Attach to Microsoft Edge`) was incorrect. Turns out this whole section was very confusing as it was talking about VS and Edge at the same time but didn't make it clear. I've tested the workflow again myself and will update accordingly.

File: microsoft-edge/visual-studio/index.md (hunk 3 above)
Comment (captainbrosset) [reply microsoft-edge/visual-studio/index.md:73]: Yep, I did test locally, which is how I was able to refresh a couple of the screenshots. The line is still 13, so this is correct.

File: microsoft-edge/visual-studio/index.md (hunk 8 above)
Comment (captainbrosset) [reply microsoft-edge/visual-studio/index.md:48]: This section wasn't very clear, I rephrased it.

File: microsoft-edge/visual-studio/index.md [hunk 11]
```diff
@@ -95,18 +92,18 @@ The example was just a minor demonstration of the functionality available in Vis
 
 ## Attach to Microsoft Edge  
 
-Previously, you had to launch Microsoft Edge from Visual Studio.  Now, you are able to attach the Visual Studio debugger to an already running instance of Microsoft Edge.  
+Previously, you launched Microsoft Edge from Visual Studio.  You can also attach the Visual Studio debugger to an already running instance of Microsoft Edge.
```
[the remark this answers is not in the document]
Comment (captainbrosset) [reply microsoft-edge/visual-studio/index.md:95]: Yeah, 2 alternative approaches that can't be used at the same time. I'll update accordingly.

**Comment by captainbrosset** [comment]
Thanks @mikehoffms, I think I addressed all of your comments. I did change the images to have red borders instead of yellow. But we might want to redo them later to have 3px borders instead of 5px (plus also to clean them up as painting the yellow regions to red left a thin yellow artefact that's not visible unless you zoom in quite a lot, so I think it's probably ok for now).

**Review by mikehoffms — APPROVED** [review]

**Pull request merged**